src/index.ts

```diff
@@ -64,18 +64,61 @@ export async function getRate(sourceToken: Token, targetToken: Token, amount: st
     return await getRateByPath(paths, amount);
 }
 
-export async function getAllPaths(sourceToken: Token, targetToken: Token) {
+export async function getAllPathsAndRates(sourceToken: Token, targetToken: Token, amount: string = '1') {
     if (sourceToken.blockchainType == 'ethereum' && targetToken.blockchainType == 'ethereum')
-        return await ethereumGetAllPaths(sourceToken.blockchainId, targetToken.blockchainId);
+        return await ethGetAllPathsAndRates(sourceToken.blockchainId, targetToken.blockchainId, amount);
     throw new Error(sourceToken.blockchainType + ' blockchain to ' + targetToken.blockchainType + ' blockchain not supported');
 }
 
+function getEthShortestPath(paths: string[][], rates: string[]): string[] {
```

Comment:
Why `Eth`? Seems like it's generic

Reply:
You're right. Fixed already (in the `redesign_and_cleanups` branch).